mono/metadata/gc.c

```diff
@@ -1019,6 +1019,8 @@ finalize_domain_objects (DomainFinalizationReq *req)
 static guint32
 finalizer_thread (gpointer unused)
 {
+	mono_profiler_finalizer_thread_start_event();
```

Comment:
What use is the finalizer thread start? This happens just once IIRC and the thread just keeps running and waiting

Reply:
The Unity profiler collects events per thread. By emitting the event on the thread it self we can create the profiler thread and all events emitted on it will automagically be associated with the thread.

Reply:
Create valid MonoProfiler object that can be passed into mono_profiler_set_profiler_events function added in this PR.

Reply:
I think we already get an event for every thread attach/detach. Is the issue that you don't know this is the finalizer thread?

Reply:
Yes, we want to explicitly signal that this is the finalizer thread.

Reply:
There is already a `mono_gc_is_finalizer_thread`. Could you expose that rather than add a new profiler callback?

Reply:
As far as I can tell profiler callbacks for are not invoked for the finalizer thread, as they are used for signaling that managed threads have started/finished. The finalizer thread is an internal "native" thread created with mono_thread_create_internal. I would prefer to avoid changing existing behavior and instead just this specific profiler callback. Also makes code on the subscriber end simpler.

Reply:
Alright, I guess we don't need a finalizer thread exit event but it would be nice for similarity to other profiler APIs.